compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -257,15 +258,16 @@ private NarrowedTypes getNarrowedTypesForBinaryOp(Map<BVarSymbol, NarrowedTypes>
             rhsTrueType = rhsFalseType = symbol.type;
         }
 
+        SymbolEnv pkgEnv = symTable.pkgEnvMap.get(this.env.enclPkg.symbol);
         BType trueType, falseType;
         if (operator == OperatorKind.AND) {
-            trueType = types.getTypeIntersection(lhsTrueType, rhsTrueType);
-            BType tmpType = types.getTypeIntersection(lhsTrueType, rhsFalseType);
+            trueType = types.getTypeIntersection(lhsTrueType, rhsTrueType, pkgEnv);
```

Comment:
Is it correct to pass the pkgEnv here? Shouldn't it be the current `env`?

For example, if this env is eventually used to create a record for an intersection, and that intersection is within a function, the owner has to be the function and not the module?

Reply:
Fixed

Reply:
I missed that, fixed it.